### PR TITLE
Video Block: refactor setting panel

### LINKED ISSUE #68341: Refactor "Settings" panel of Video block to use `ToolsPanel` instead of `PanelBody`
Part of #67813 

Convert the "Settings" panel of the Video block to use the `ToolsPanel` instead of `PanelBody`

### PR DESCRIPTION
**Pull request by up1512001**
## What?
Splitting from https://github.com/WordPress/gutenberg/pull/66946

## Why?
While adding `DimensionsTool` to video block it appears it needs to be inside `ToolsPanel` component instead of `PanelBody`.

## Testing Instructions
1. Add video block
2. open setting panel 

## Screenshots or screencast 

https://github.com/user-attachments/assets/db6ebd1b-643b-4590-a0ff-7ccd973a3975

### CODE REVIEW

**Review by aaronrobertshaw — COMMENTED** [review]
Thanks for spinning up this PR @up1512001 🚀 

Before we commit to refactoring the panel. As @t-hamano has proposed on your original PR, it would be good to see what's involved in the [block support approach](https://github.com/WordPress/gutenberg/pull/67047) first.

My understanding of your original PR was that your use case needed an additional option (contain?) that the existing aspect ratio block support didn't provide? Can you give https://github.com/WordPress/gutenberg/pull/67047 a test and review to see if it does meet your needs? 

Even if we choose to adopt the block support instead of https://github.com/WordPress/gutenberg/pull/66946, this refactor of the panel might still be desirable to bring the block inline with the Cover and Image blocks. We can make that call later though once we get the functionality we want.

**Comment by t-hamano** [comment]
I left a comment [here](https://github.com/WordPress/gutenberg/issues/60911#issuecomment-2495452552) to discuss what approach would be ideal to resolve #60911.

**Comment by up1512001** [comment]
@t-hamano As we are now moving towards `ToolsPanel` for many blocks it would be good to implement `aspect-ratio` & `scale` through `DimensionsTool` component. 
let me know if we should proceed and first convert the video block Panel component to ToolsPanel.
This is a reference https://github.com/WordPress/gutenberg/issues/67813

File: packages/block-library/src/video/edit.js [hunk 1]
```diff
@@ -214,63 +200,31 @@ function VideoEdit( {
 				</>
 			) }
 			<InspectorControls>
-				<PanelBody title={ __( 'Settings' ) }>
+				<ToolsPanel
+					label={ __( 'Settings' ) }
+					resetAll={ () => {
+						setAttributes( {
+							autoplay: false,
+							controls: false,
```
Comment (t-hamano) [inline packages/block-library/src/video/edit.js]: The default value for the Playback controls (`controls` attribute) is `true`.

File: packages/block-library/src/video/edit-common-settings.js [hunk 2]
```diff
@@ -47,50 +51,104 @@ const VideoSettings = ( { setAttributes, attributes } ) => {
...
+			<ToolsPanelItem
 				label={ __( 'Playback controls' ) }
-				onChange={ toggleFactory.controls }
-				checked={ !! controls }
-			/>
-			<ToggleControl
-				__nextHasNoMarginBottom
-				/* translators: Setting to play videos within the webpage on mobile browsers rather than opening in a fullscreen player. */
+				isShownByDefault
+				hasValue={ () => !! controls }
```
Comment (t-hamano) [inline packages/block-library/src/video/edit-common-settings.js]:
```suggestion
				hasValue={ () => ! controls }
```

The default value for the Playback controls (`controls` attribute) is `true`. Therefore, "having a value" means that it is not the default value, i.e. `false`.

**Review by t-hamano — COMMENTED** [review]
@up1512001 Sorry for the late reply.

> let me know if we should proceed and first convert the video block Panel component to ToolsPanel.

Yes, it's worth moving forward with this PR regardless of how we move forward with #66946.

It's mostly working as expected, but I'd be happy if you could rebase just to be sure.

**Comment by up1512001** [comment]
@t-hamano updated the default state for playback controls to `true` by default & rebase the PR.

https://github.com/user-attachments/assets/5d796c37-daf4-4492-8ffc-3a535da1ea32

**Review by t-hamano — APPROVED** [review]
LGTM 👍

**Pull request merged**